apps/ecommerce/frontend/src/components/Config/ConfigPage.tsx

```diff
@@ -83,16 +83,11 @@ const ConfigPage = () => {
 
   const checkCredentials = async (baseUrl: string) => {
     const url = new URL(`${baseUrl}/healthcheck`);
-    fetchWithSignedRequest(
-      url,
-      sdk.ids.app,
-      cma,
-      'POST',
-      {
-        'X-Contentful-Data-Provider': sdk.parameters.instance.provider,
-      },
-      { domain: parameters.apiEndpoint, storefrontAccessToken: parameters.storefrontAccessToken }
-    )
+    fetchWithSignedRequest(url, sdk.ids.app, cma, 'POST', {
+      'X-Contentful-Data-Provider': sdk.parameters.instance.provider,
+      'x-contentful-shopify-domain': parameters.apiEndpoint,
```

Comment:
we should *not* be passing this through as a header. it should be coming from the [provider configs](https://github.com/contentful/apps/blob/master/apps/ecommerce/lambda/src/mocks/providerConfigs.ts) that @whitelisab added to avoid introducing man-in-the-middle/SSRF attacks

Reply:
Not sure I follow, how do the values get passed to the endpoint via the provider configs? Should `apiEndpoint/storefrontAccessToken` be body params?

Reply:
to be clear, i'm referring to the apiEndpoint – URL things shouldn't be coming from the frontend because someone could make a request with a bad actor endpoint like `iscrapecredentials.com` and then get the access tokens of our customers.

let's start a thread in the eng chat to discuss further